src/components/Bakeries/Schedule/ScheduleContainer.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router";
+import Schedule from "./Schedule";
+import { mainApi } from "../../../api/api.js";
+
+const ScheduleContainer = (props) => {
+  let param = useParams();
+  const [schedule, setSchedule] = useState([]);
+  const [name, setName] = useState("");
+  const [err, setError] = useState(false);
+
+  useEffect(() => {
+    (async function () {
+      let map = new Map(),
+        schedule = [],
+        prevDayName = "",
+        list = await mainApi.getBakery(param.name);
+
+      if (list.message) {
+        setError(true);
+      } else {
+        let upperFirst = (day) =>
+          day.split("")[0].toLocaleUpperCase() + day.slice(1);
+
+        for (let day in list.schedule) {
+          if (prevDayName === "") prevDayName = day;
+          if (list.schedule[day].day_off === false) {
+            if (day === "monday") map[prevDayName] = day;
+            if (
+              JSON.stringify(list.schedule[day]) ===
+              JSON.stringify(list.schedule[prevDayName])
+            ) {
+              map[prevDayName] = day;
+            } else {
+              map[day] = day;
+              prevDayName = day;
+            }
+          } else if (list.schedule[day].day_off === true) {
+            map[day] = "close";
+            prevDayName = "";
+          }
+        }
+
+        for (let day in map) {
```

Comment:
I suggest to simplify that logic a bit:
1) make such long expressions shorter `list.schedule[day].order_before`
2) you have 4 different string generators, while they in reality are the same: 
- move day/days in the variable
- return the generated string instead of pushing in 4 different places (make an extra method)

Reply:
I've done it in my own and new way) Hope it's fine and more better)